v4_MIGRATION.md

```diff
@@ -0,0 +1,95 @@
+# Android Popup Bridge v4 (Beta) Migration Guide
+
+See the [CHANGELOG](/CHANGELOG.md) for a complete list of changes. This migration guide outlines the basics for updating your popup bridge integration from v3 to v4.
+
+## Setup
+
+Add the library to your dependencies in your `build.gradle`:
+
+```groovy
+dependencies {
+  implementation 'com.braintreepayments.api:popup-bridge:4.0.0-beta1'
+}
+```
+
+Then, add an `intent-filter` in the `AndroidManifest.xml` to your deep link destination activity:
+
+```xml
+<activity android:name="com.company.app.MyPopupBridgeActivity">
+    ...
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW"/>
+        <data android:scheme="my-custom-url-scheme"/>
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+    </intent-filter>
+</activity>
+```
+
+**Note**: The scheme you define must use all lowercase letters. This is due to [scheme matching on the Android framework being case sensitive, expecting lower case](https://developer.android.com/guide/topics/manifest/data-element#scheme).
+
+If these requirements are not met, an error will be reported to the `PopupBridgeErrorListener`, if registered.
```

Comment:
I think since this information also applies to v3 (and doesn't represent a change that came in v4), we can instead move this to the README. This seems more detailed than the current instructions in the README.

Reply:
Actually - I think this stems from a difference in how we wrote the iOS migration guide and how this one is written. For [iOS](https://github.com/braintree/braintree_ios/blob/master/V5_MIGRATION.md) we kept the migration guide specific to assuming the merchant already has a working v3 integration and only listed the additional (or breaking) changes they would need to add.

Reply:
Yeah we can review the iOS guide again for consistency. There is a _slight_ difference in `v4` though. In `v3`, `com.braintreepayments.popupbridge.PopupBridgeActivity` is owned by the Braintree SDK. In `v4` docs, we mention `MyPopupBridgeActivity` to signal to the merchant that they now own the deep-link destination activity.

Reply:
+1 to having the migration guide just list the required changes for someone who already has a working v3 integration

Reply:
Updated it a bit to mention `v3`. I think it's a fairly straightforward migration but we can always enhance the docs more before GA if we see a need.